src/renderer/components/stake/AddStake/AddStake.tsx

```diff
@@ -14,113 +23,126 @@ type Props = {
   runeAsset: Asset
   assetPrice: BigNumber
   runePrice: BigNumber
-  assetAmount: BaseAmount
-  runeAmount: BaseAmount
-  priceIndex?: PriceDataIndex
+  assetBalance: BaseAmount
+  runeBalance: BaseAmount
   unit?: string
   assetData?: AssetPair[]
   className?: string
-  onStake: (stakeAmount: BigNumber, asset: Asset) => void
+  onStake: (stakeData: { asset: Asset; runeAsset: Asset; assetStake: BaseAmount; runeStake: BaseAmount }) => void
+  onChangeAsset: (asset: Asset) => void
+  disabled?: boolean
+  poolData: PoolData
 }
 
 export const AddStake: React.FC<Props> = ({
   asset,
   runeAsset,
   assetPrice,
   runePrice,
-  assetAmount: assetAmountProp,
-  runeAmount,
+  assetBalance,
+  runeBalance,
   className,
   assetData,
   unit,
-  onStake
+  onStake,
+  onChangeAsset,
+  disabled,
+  poolData
 }) => {
   const intl = useIntl()
-  /**
-   * Hold stakeAmount as amount of runes to stake
-   */
-  const [stakeAmount, setStakeAmount] = useState(baseAmount(0))
+  const [stakeRunes, setStakeRunes] = useState<AssetAmount>(ZERO_ASSET_AMOUNT)
+  const [stakeAssets, setStakeAssets] = useState<AssetAmount>(ZERO_ASSET_AMOUNT)
+
+  const runeSelect = useMemo(() => {
```

Comment:
[nitpick] Suggestion for wording:
- `stakeRunes` -> `runeToStakeAmount`
- `stakeRunes` -> `assetToStakeAmount`
- `runeSelect` -> `runeToStakeBaseAmount`